### PR TITLE
Windows: support getting the thread name

### PR DESCRIPTION
**Pull request by RalfJung**
Also organize the thread name tests a bit.

### CODE REVIEW

File: src/shims/windows/foreign_items.rs [hunk 1]
```diff
@@ -412,7 +422,31 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
...
+
+                let handle = this.read_scalar(handle)?;
+                let name_ptr = this.deref_pointer(name_ptr)?; // the pointer where we should store the ptr to the name
+
+                let thread = match Handle::from_scalar(handle, this)? {
+                    Some(Handle::Thread(thread)) => thread,
+                    Some(Handle::Pseudo(PseudoHandle::CurrentThread)) => this.get_active_thread(),
+                    _ => this.invalid_handle("SetThreadDescription")?,
+                };
+                // FIXME what is the right default thread name? Or should we raise an error?
```
Comment (ChrisDenton) [inline src/shims/windows/foreign_items.rs:442]: The right default would be an empty string. Though in std that would be changed to `<unnamed>` anyways.

Comment (RalfJung) [reply src/shims/windows/foreign_items.rs:442]: So on actual Windows it succeeds and returns an empty string when no name was that?
That's easy enough to do. Is this documented anywhere?

Yeah I know this is not observable on std.

Comment (ChrisDenton) [reply src/shims/windows/foreign_items.rs:442]: > Is this documented anywhere?

No.

**Pull request merged**